packages/url_launcher/url_launcher_web/lib/src/link.dart

```diff
@@ -70,88 +110,273 @@ class WebLinkDelegateState extends State<WebLinkDelegate> {
+            return PlatformViewSurface(
+              controller: controller,
+              gestureRecognizers: const <Factory<
+                  OneSequenceGestureRecognizer>>{},
+              hitTestBehavior: PlatformViewHitTestBehavior.transparent,
+            );
+          },
+        ),
+      ),
+    );
+  }
 }
 
 final JSAny _useCapture = <String, Object>{'capture': true}.jsify()!;
 
+/// Signature for the function that triggers a link.
+typedef TriggerLinkCallback = void Function(
+    int viewId, html.MouseEvent? mouseEvent);
+
+/// Keeps track of the signals required to trigger a link.
+///
+/// Currently, the signals required are:
+///
+/// 1. A FollowLink signal. This signal indicates that a hit test for the link
+///    was registered on the app/framework side.
+///
+/// 2. A DOM event signal. This signal indicates that a click or keyboard event
+///    was received on the link's corresponding DOM element.
+///
+/// These signals can arrive in any order depending on how the user triggers
+/// the link.
+///
+/// Each signal may be accompanied by a view ID. The view IDs, when present,
+/// must match in order for the trigger to be considered valid.
+///
+/// The signals are reset after a certain delay to prevent them from getting
+/// stale. The delay is specified by [staleTimeout].
+class LinkTriggerSignals {
+  /// Creates a [LinkTriggerSignals] instance that calls [triggerLink] when all
+  /// the signals are received within a [staleTimeout] duration.
+  LinkTriggerSignals({
+    required this.triggerLink,
+    required this.staleTimeout,
+  });
+
+  /// The function to be called when all signals have been received and the link
+  /// is ready to be triggered.
+  final TriggerLinkCallback triggerLink;
+
+  /// Specifies the duration after which the signals are considered stale.
+  ///
+  /// Signals have to arrive within [staleTimeout] duration between them to be
+  /// considered valid. If they don't, the signals are reset.
+  final Duration staleTimeout;
+
+  bool get _hasAllSignals => _hasFollowLink && _hasDomEvent;
+
+  int? get _viewId {
+    assert(!isViewIdMismatched);
+    return _viewIdFromFollowLink ?? _viewIdFromDomEvent;
+  }
+
+  /// Triggers the link if all signals are ready and there's no view ID
+  /// mismatch.
+  ///
+  /// If the view IDs from the signals are mismatched, the signals are reset and
+  /// the browser is prevented from navigating.
+  ///
+  /// If the signals are ready, the link is triggered and the signals are reset.
+  void triggerLinkIfReady() {
+    if (isViewIdMismatched) {
+      // When view IDs from signals are mismatched, let's reset the signals and
+      // prevent the browser from navigating.
+      _mouseEvent?.preventDefault();
+      reset();
+      return;
+    }
+
+    if (_hasAllSignals) {
+      triggerLink(_viewId!, _mouseEvent);
+      reset();
+    }
+  }
+
+  /// Acknowledges a FollowLink signal from [viewId].
+  void acknowledgeFollowLink({required int viewId}) {
```

Comment:
Let's go with one of `on`, `accept`, `receive`, or `process`. These are more common for methods that receive events.

Reply:
Renaming to `on`, I like it, and it's the shortest :)